zkevm-circuits/src/root_circuit.rs

```diff
@@ -170,18 +244,183 @@ where
+                        let (zero_const, one_const, total_chunk_const) = {
+                            let zero_const = loader
+                                .scalar_chip()
+                                .assign_constant(&mut loader.ctx_mut(), M::Scalar::from(0))
+                                .unwrap();
+                            let one_const = loader
+                                .scalar_chip()
+                                .assign_constant(&mut loader.ctx_mut(), M::Scalar::from(1))
+                                .unwrap();
+                            let total_chunk_const = loader
+                                .scalar_chip()
+                                .assign_constant(&mut loader.ctx_mut(), M::Scalar::from(1))
+                                .unwrap();
+                            (zero_const, one_const, total_chunk_const)
+                        };
+
+                        // `first.sc_rwtable_row_prev_fingerprint ==
+                        // first.ec_rwtable_row_prev_fingerprint` will be checked inside circuit
+                        vec![
+                            // chunk ctx
+                            (first_chunk.chunk_index.assigned(), &zero_const),
+                            (first_chunk.total_chunk.assigned(), &total_chunk_const),
+                            // rwc
+                            (first_chunk.initial_rwc.assigned(), &one_const),
+                            // constraint permutation fingerprint
+                            // challenge: alpha
+                            (first_chunk.sc_permu_alpha.assigned(), &alpha.assigned()),
+                            (first_chunk.ec_permu_alpha.assigned(), &alpha.assigned()),
+                            // challenge: gamma
+                            (first_chunk.sc_permu_gamma.assigned(), &gamma.assigned()),
+                            (first_chunk.ec_permu_gamma.assigned(), &gamma.assigned()),
+                            // fingerprint
+                            (
+                                first_chunk.ec_rwtable_prev_fingerprint.assigned(),
+                                &one_const,
+                            ),
+                            (
+                                first_chunk.sc_rwtable_prev_fingerprint.assigned(),
+                                &one_const,
+                            ),
+                        ]
+                        .iter()
+                        .for_each(|(a, b)| {
+                            loader
+                                .scalar_chip()
+                                .assert_equal(&mut loader.ctx_mut(), a, b)
+                                .unwrap();
+                        });
+
+                        first_chunk
+                    })
+                    .expect("error");
+
+                // constraint consistency between chunk
+                let _ = supercircuit_instances.iter().tuple_windows().inspect(
```

Comment:
`inspect` is also lazy so the closure is not called afais, so perhaps using `for_each` here?

Reply:
Thanks for point out this. Fixed by `for_each` which is more semantics correct